### PR TITLE
Use import instead of require in with-apollo example

### PR DESCRIPTION
**Pull request by HaNdTriX**
more info: **https://github.com/zeit/next.js/pull/8556#pullrequestreview-281583900**

### CODE REVIEW

File: examples/with-apollo/lib/apollo.js [hunk 1]
```diff
@@ -56,7 +56,8 @@ export function withApollo (PageComponent, { ssr = true } = {}) {
 
         try {
           // Run all GraphQL queries
-          await require('@apollo/react-ssr').getDataFromTree(
+          const { getDataFromTree } = await import('@apollo/react-ssr')
```
Comment (Timer) [inline examples/with-apollo/lib/apollo.js]: Should this be the following?
```suggestion
          const { getDataFromTree } = await import('@apollo/react-ssr').then(e => e.default)
```

Comment (HaNdTriX) [reply examples/with-apollo/lib/apollo.js]: Afaik they use names exports: https://github.com/apollographql/react-apollo/blob/master/packages/ssr/src/index.ts#L1

I am testing every change I do in production and development mode. But let me take a deeper look. I have never used `async imports` in node context. Might be interesting what this compiles to.

Comment (HaNdTriX) [reply examples/with-apollo/lib/apollo.js]: A bit off topic 🤔

#### Using `async import(...)`

```js
const {
  getDataFromTree
} = await Promise.resolve(/* import() */).then(__webpack_require__.t.bind(null, "7fVs", 7));
```

#### Using `require(...)`

```js
const {
  getDataFromTree
} = __webpack_require__("7fVs");
```

**Review by lfades — APPROVED** [review]
@HaNdTriX Thank you 🥇

File: examples/with-apollo/lib/apollo.js (hunk 1 above)
Comment (msheakoski) [reply examples/with-apollo/lib/apollo.js]: Why not just import this at the top of the file? What is the purpose of making it dynamic?

Comment (HaNdTriX) [reply examples/with-apollo/lib/apollo.js]: This module will be shipped to the client if you import it at the top of the file, even though it is only needed on the server.

Keep in mind that the code of your next application will run in multiple different contexts (node.js & Browser).

Comment (msheakoski) [reply examples/with-apollo/lib/apollo.js]: Interesting, that makes sense now. Thank you for clarifying!

**Pull request merged**